### ISSUE TITLE
Cannot install postgreSQL extension pgTAP 1.3.1

### ISSUE THREAD

**Issue by datafoo** [NixOS/nixpkgs#272232]
### Describe the bug
I cannot install the pgTAP 1.3.1 extension

### Steps To Reproduce
1. Pin nixpkgs to commit introducing pgTAP 1.3.1
    ```nix
    nixpkgs-unstable = { url = github:NixOS/nixpkgs?rev=e803ddcaf5a947e8fab28945758c9d011f304ba4; };
    ```
2. Use PostgreSQL 15 with the pgTAP extension
    ```nix
    postgresql = pkgsUnstable.postgresql_15.withPackages (ps: [
        ps.pgtap
    ]);
    ```
3. Create a PostgreSQL cluster
4. Run:
    ```sql
    CREATE EXTENSION pgtap;
    ```

### Actual behavior
The extension is not installed and an error is shown:
```log
ERROR:  could not access file "$libdir/pgtap": No such file or directory
```

### Expected behavior
The extension should be installed.

### Additional context
@marsam: pgTAP 1.3.1 was [released on 2023-09-24](https://github.com/theory/pgtap/releases/tag/v1.3.1), yet you [committed](https://github.com/NixOS/nixpkgs/commit/6f9e07dd45f0aed349ef90ddff996f05d7df3974) the updated from 1.3.0 to 1.3.1 on 2023-09-22, 2 days before the release. How is this possible?

### Notify maintainers

@marsam, PR #259734

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.64, NixOS, 23.11 (Tapir), 23.11.1209.933d7dc15509`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(me): `"home-manager-23.05.tar.gz"`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by SuperSandro2000** [comment]
> 2\. Use PostgreSQL 15 with the pgTAP extension

Please try the services.postgresql.extraPlugins option. It could be ondoing your override.

**Comment by SuperSandro2000** [comment]
> @r-ryantm I see you contributed to this extension as well. Do you have any feedback to provide?

That is a bot, it will not answer 😅

**Comment by marsam** [comment]
Sorry for the delay. I'm awfully busy with real life matters. I just now had time to open my notifications.

Sorry, my bad, I didn't test this extension properly. I'm reverting it https://github.com/NixOS/nixpkgs/issues/272232 
I'll take a proper look after holidays.

**Comment by theory** [comment]
I suspect your packaging stuff will need to be updated to add installation of the shared library added to v1.3.1. Just copy `pgtap.so` (I assume the extension is `.so`) from the `lib` directory after building pgTAP.

See theory/pgtap#315 for discussion about adding the shared library.

**Comment by theory** [comment]
FYI I've just released v1.3.2, which removes the shared object library from pgTAP. You'll have to reverse 276486 when you update the package. Apologies for the hassle.
